consensus/src/zk.rs

```diff
@@ -0,0 +1,179 @@
+use std::process;
+use std::time::Duration;
+use std::thread::{self, Thread};
+
+use serde::Serialize;
+use serde::de::DeserializeOwned;
+use serde_json;
+use zookeeper::{Acl, CreateMode, KeeperState, WatchedEvent, Watcher, ZkError, ZooKeeper};
+
+use Authority;
+use Epoch;
+use CONTROLLER_KEY;
+
+struct EventWatcher;
+impl Watcher for EventWatcher {
+    fn handle(&self, e: WatchedEvent) {
+        if e.keeper_state != KeeperState::SyncConnected {
+            eprintln!("Lost connection to ZooKeeper! Aborting");
+            process::abort();
+        }
```

Comment:
OOI, what's the purpose of `EventWatcher`? It doesn't do anything except check that the event's `KeeperState` is still `SyncConnected`.

EDIT: looks like it's used as an error handler for connecting to Zookeeper. Perhaps we should call it `ConnectSuccessWatcher` or something similar? I was misled into believing it was a more general event handler.

Reply:
This function is called with any out of band Zookeeper events that don't have designated event handlers specified for them. I'm currently only using it to detect if the connection to Zookeeper was lost and thus any associated leases expired (It is kind of unclear to to actually handle this case, so for now I just abort instead).

As a side note I think a failed connection would actually trigger an error earlier.